### DISCUSSION TITLE
Capture video in Github shareable format

### DISCUSSION THREAD

**Discussion by Torinde** [joncampbell123/dosbox-x#4247] [Q&A] [answered]
For an Issue I tried capturing video of what's happening in DOSbox:
![image](https://user-images.githubusercontent.com/97228894/236662730-af182bd1-c7e3-4098-a8d4-55e9e3f1caf6.png)

I got an AVI file (unplayable by Windows integrated players) that can't be uploaded to Github:
![image](https://user-images.githubusercontent.com/97228894/236662843-4076bad1-474b-4203-8907-41dbc93a4f2f.png)

I've seen others posting such videos here, so what am I missing? Is everybody converting these videos manually to Github-supported format - or there is some setting that I need to change?

**Comment by Torinde** [comment]
I tried adding `capture format            = mpegts-h264` to [dosbox] section of .conf, but menu remained as above (e.g. AVI+ZMBV). Does this work only on Linux/macOS? Or I need to enable it in a different way?

**Comment by rderooy** [comment, marked as answer]
I think it is grayed out because the dosbox-x binary was not compiled with ffmpeg.

**Reply by Torinde** [reply]
Most probably you're correct - .conf file: `capture format: Capture format to use when capturing video. The availability of the format depends on how DOSBox-X was compiled.`

Pity, it means release builds can do only AVI+ZMBV+PCM.

**Comment by Torinde** [comment]
So, 
- Github [supports](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/attaching-files) MOV, MP4, WebM containers - codec support is browser specific, they recommend H.264
- Github doesn't support AVI and browsers most probably don't support ZMBV.

**Comment by joncampbell123** [comment]
I also want to point out that AVI and MTS (MPEG transport stream) have the benefit that should DOSBox-X crash, the incomplete capture file is either playable or easily recoverable (VLC player will handily offer to rebuild the index of an incomplete AVI for example).

When FFMPEG is involved I could provide MP4 recording, but understand that MP4, like QuickTime, is one of those formats where if the program writing it crashes, the incomplete file is garbage and unusable.
